app/styleguide/page.tsx

```diff
@@ -1037,23 +1059,189 @@ function Page() {
+            <dl>
+              <dt><code>--layout-gap</code></dt>
+              <dd>
+                This container uses flexbox, and this variable sets
+                the <code>gap</code> css property. Defaults to <code>1em</code>
+              </dd>
+            </dl>
+
+            <h2>Sidebar Container (Composite)</h2>
+            <p>This is a composite layout component that contains a sidebar
+              alongside a content container.</p>
+            <p><code>LayoutContainer &gt; LayoutWithSidebar</code></p>
+            <p><code>ContentContainer &gt; SidebarContainer</code></p>
+            <ToolbarContainer>
+              <Button onPress={toggleSidebar}>Toggle Sidebar</Button>
+            </ToolbarContainer>
+            <LayoutWithSidebar className="sg-sidebar">
+              <ContentContainer>
+                <h3>Layout With Sidebar</h3>
+                <p>This is the primary content for the slider component</p>
+
+                <div><pre><code>
+                  {`<LayoutWithSidebar>
+  <ContentContainer> ... </ContentContainer>
+  <SidebarContainer> ... </SidebarContainer>
+</LayoutWithSidebar>
+`}
+                </code></pre></div>
+              </ContentContainer>
+
+              <SidebarContainer isOpen={sidebarOpen}>
+                <p>This is the sidebar for the component</p>
+              </SidebarContainer>
+            </LayoutWithSidebar>
 
-            <h2>Layout</h2>
-            <p>TBD ...</p>
+            <p>
+              <code>LayoutWithSidebar</code> &nbsp;
+              can have a <code>ToolbarContainer</code> in addition to the sidebar.
+            </p>
+            <p>Note how in the previous example, the toolbar that demonstrates
+              the sidebar position and state, is <em>outside</em> &nbsp;
+              the <code>LayoutWithSidebar</code>, where in the example below, it's
+              part of it.</p>
+
+            <LayoutWithSidebar className="sg-sidebar">
+              <ToolbarContainer>
+                <p>This is a toolbar inside <code>LayoutWithSidebar</code></p>
+              </ToolbarContainer>
+
+              <ContentContainer>
+                <h3>Layout With Sidebar</h3>
+                <p>This is the primary content for the slider component</p>
+                <div><pre><code>
+                  {`<LayoutWithSidebar>
+  <ToolbarContainer> ... </ToolbarContainer>
+  <SidebarContainer> ... </SidebarContainer>
+  <ContentContainer> ... </ContentContainer>
+</LayoutWithSidebar>
+`}
+                </code></pre></div>
+              </ContentContainer>
+
+              <SidebarContainer>
+                <p>This is the sidebar for the slider component</p>
+              </SidebarContainer>
+            </LayoutWithSidebar>
+
+            <p>Example with a drawer instead of a sidebar</p>
+
+            <ToolbarContainer>
+              <Button onPress={toggleDrawer}>Toggle Drawer</Button>
```

Comment:
When I open the sidebar, I'm not seeing the sidebar get focus. And when I close the sidebar, should the focus go back to where the user was last focused on the main page?

Reply:
Unlike the DrawerPanel, the SidebarPanel don't change focus when the SidebarPanel state changes. This is due to how sidebar panels are used. I'm open to change this though, but this probably deserves a tiny bit of discussion together with @fraserclark .

DrawerPanel on the other hand do receive focus when opened, and focus goes back to previous focus when closed.